core/modules/image/image.field.inc

```diff
@@ -755,26 +804,42 @@ function image_field_formatter_view($entity_type, $entity, $field, $instance, $l
   }
 
   foreach ($items as $delta => $item) {
-    // Add class for floating the image
-    if (!empty($display['settings']['image_float'])) {
-      $item['attributes']['class'][] = 'align-' . $display['settings']['image_float'];
-    }
+    if (!empty($item)) {
+      // Add class for floating the image.
+      if (!empty($display['settings']['image_float'])) {
+        $item['attributes']['class'][] = 'align-' . $display['settings']['image_float'];
+      }
 
-    if (!empty($display['settings']['image_load'])) {
-      // Although available in Chromium, the 'auto' value is not mentioned in
-      // the specification. Since it may be subject to change, we recommend not
-      // to use it until it gets officially included.
-      if ($display['settings']['image_load'] != 'auto') {
-        $item['attributes']['loading'] = $display['settings']['image_load'];
+      if (!empty($display['settings']['image_load'])) {
+        // Although available in Chromium, the 'auto' value is not mentioned in
+        // the specification. Since it may be subject to change, we recommend
+        // not to use it until it gets officially included.
+        if ($display['settings']['image_load'] != 'auto') {
+          $item['attributes']['loading'] = $display['settings']['image_load'];
+        }
+      }
+
+      if (isset($link_file)) {
+        $uri = array(
+          'path' => file_create_url($item['uri']),
+          'options' => array(),
+        );
       }
-    }
 
-    if (isset($link_file)) {
-      $uri = array(
-        'path' => file_create_url($item['uri']),
-        'options' => array(),
-      );
+      $is_svg = image_is_svg($item['uri']);
+
+      // Add custom attributes only for SVG images.
+      if ($is_svg) {
+        $image_svg_settings = $display['settings']['image_svg_settings'];
+
+        foreach ($image_svg_settings as $attribute => $attribute_value) {
+          if (!empty($attribute)) {
+            $item[$attribute] = (int) $attribute_value;
```

Comment:
I haven't actually tested this, but from reading the code it seems that this would add the attributes with only numeric values. Shouldn't we be adding `px` suffix here?

Reply:
No, `px` not needed. This results in (for example) `height='500'` attribute which defaults to `500px`.